### PR TITLE
Add subsystem class

### LINKED ISSUE #28: Add subsystem class
We should at least allow for something like:

```kotlin
fun defaultCommand = ArcadeDrive().start()
```

instead of WPILib's

```kotlin
override fun initDefaultCommand() {
    ArcadeDrive().start()
}
```

### PR DESCRIPTION
**Pull request by andrewda**
Closes #28

Literally have no clue if this will work, but it's compiling so...

### CODE REVIEW

File: core/src/main/kotlin/org/sertain/command/Subsystem.kt [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+package org.sertain.command
+
+import org.sertain.RobotLifecycle
+
+private typealias WpiLibSubsystem = edu.wpi.first.wpilibj.command.Subsystem
+
+/**
+ * @see edu.wpi.first.wpilibj.command.Subsystem
+ */
+public abstract class Subsystem : WpiLibSubsystem(), RobotLifecycle
```
Comment (SUPERCILEX) [inline core/src/main/kotlin/org/sertain/command/Subsystem.kt]: In `onCreate()`, you gotta add a listener for the lifecycle.

Comment (andrewda) [reply core/src/main/kotlin/org/sertain/command/Subsystem.kt]: Why `onCreate()` and not `init` like in Command?

Comment (andrewda) [reply core/src/main/kotlin/org/sertain/command/Subsystem.kt]: I did `init` just to mimic Command for now. Will change it to `onCreate()` if that's how it should be done.

Comment (SUPERCILEX) [reply core/src/main/kotlin/org/sertain/command/Subsystem.kt]: Cuz this is a library and people will be extending the class. Example: someone adds a field that they use in one of the callbacks -> this callback is added in the super constructor before the field has been initialized -> NPE.

**Pull request merged**